libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/UIColor+SFColors.m

```diff
@@ -37,18 +37,35 @@ +(UIColor *) salesforceBlueColor {
 
 + (UIColor *)colorFromHexValue:(NSString *)hexString {
     UIColor *color = nil;
+    hexString = [[self class] sixDigitHexFromString:hexString];
     if ([hexString length] > 0) {
         unsigned rgbValue = 0;
         NSScanner *scanner = [NSScanner scannerWithString:hexString];
-        if ([hexString rangeOfString:@"#"].location != NSNotFound) {
-            [scanner setScanLocation:1]; // bypass '#' character
-        }
         [scanner scanHexInt:&rgbValue];
         color = [UIColor colorWithRed:((rgbValue & 0xFF0000) >> 16)/255.0 green:((rgbValue & 0xFF00) >> 8)/255.0 blue:(rgbValue & 0xFF)/255.0 alpha:1.0];
     }
     return color;
 }
 
++ (NSString *)sixDigitHexFromString:(NSString *)hexString {
```

Comment:
Except for legacy definitions, let's prefix category extension definitions on Cocoa classes with `msdk`, to avoid any potential naming conflicts with other libraries.